samples/ProtectedMCPServer/Program.cs

```diff
@@ -0,0 +1,104 @@
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.IdentityModel.Tokens;
+using ModelContextProtocol.AspNetCore.Authentication;
+using ModelContextProtocol.Authentication;
+using ProtectedMCPServer.Tools;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+
+var builder = WebApplication.CreateBuilder(args);
+
+var serverUrl = "http://localhost:7071/";
+var tenantId = "a2213e1c-e51e-4304-9a0d-effe57f31655";
+var instance = "https://login.microsoftonline.com/";
+
+builder.Services.AddAuthentication(options =>
+{
+    options.DefaultChallengeScheme = McpAuthenticationDefaults.AuthenticationScheme;
+    options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
+})
+.AddJwtBearer(options =>
+{
+    options.Authority = $"{instance}{tenantId}/v2.0";
+    options.TokenValidationParameters = new TokenValidationParameters
+    {
+        ValidateIssuer = true,
+        ValidateAudience = true,
+        ValidateLifetime = true,
+        ValidateIssuerSigningKey = true,
+        ValidAudience = "167b4284-3f92-4436-92ed-38b38f83ae08",
+        ValidIssuer = $"{instance}{tenantId}/v2.0",
+        NameClaimType = "name",
+        RoleClaimType = "roles"
+    };
+
+    options.MetadataAddress = $"{instance}{tenantId}/v2.0/.well-known/openid-configuration";
+
+    options.Events = new JwtBearerEvents
+    {
+        OnTokenValidated = context =>
+        {
+            var name = context.Principal?.Identity?.Name ?? "unknown";
+            var email = context.Principal?.FindFirstValue("preferred_username") ?? "unknown";
+            Console.WriteLine($"Token validated for: {name} ({email})");
+            return Task.CompletedTask;
+        },
+        OnAuthenticationFailed = context =>
+        {
+            Console.WriteLine($"Authentication failed: {context.Exception.Message}");
+            return Task.CompletedTask;
+        },
+        OnChallenge = context =>
+        {
+            Console.WriteLine($"Challenging client to authenticate with Entra ID");
+            return Task.CompletedTask;
+        }
+    };
+})
+.AddMcp(options =>
+{
+    options.ProtectedResourceMetadataProvider = context => 
+    {
+        var metadata = new ProtectedResourceMetadata
+        {
+            Resource = new Uri("http://localhost"),
+            BearerMethodsSupported = { "header" },
+            ResourceDocumentation = new Uri("https://docs.example.com/api/weather"),
+            AuthorizationServers = { new Uri($"{instance}{tenantId}/v2.0") }
```

Comment:
```suggestion
            AuthorizationServers = { new Uri($"{instance}{tenantId}/oauth/v2.0") }
```
shouldn't it be this?